.devcontainer/setup.sh

```diff
@@ -0,0 +1,8 @@
+#!/usr/bin/env bash
+
+set -euo pipefail
+IFS=$'\n\t'
+set -vx
+
+mix local.hex --force
+mix deps.get
```

Comment:
It _seems like_ these are the two most relevant commands to run immediately after creating the container.

Without, an initial run of `mix test` prompted me to run `mix local.hex --force` and an initial run of `mix docs` required running `mix deps.get`.

For a library like this, are there other helpful commands that should be run after container creation?

Reply:
Quick note here in case it's instructive: Elixir's build tool `mix` has a concept of [environments](https://hexdocs.pm/mix/1.13.4/Mix.html#module-environments). Briefly, each environment is a completely separate build of the project but with slight configuration changes.

`:compare_chain` itself has no dependencies in the `:prod` or `:test` environments. But it does have the `:ex_doc` dependency in the `:dev` environment. We can see these two facts from how `:compare_chain` declares its dependencies in `mix.exs`:

```elixir
  defp deps do
    [
      {:ex_doc, "~> 0.31", only: :dev, runtime: false}
                         # ^^^^^^^^^^
    ]
  end
```

This is why you can run `mix test` (which runs in the `:test` environment) without needing `mix deps.get`, but you do need it to run `mix docs` (which runs in the `:dev` environment).

Reply:
> For a library like this, are there other helpful commands that should be run after container creation?

No I think what's there is good!